src/grid.rs

```diff
@@ -1,38 +1,29 @@
 use macroquad::rand;
 
-pub const GRID_WIDTH: usize = 32;
-pub const GRID_HEIGHT: usize = 32;
-pub const CELL_SIZE: f32 = 30.0;
-
-#[derive(Copy, Clone, PartialEq, Default, Debug)]
+#[derive(Copy, Clone, PartialEq, Debug)]
 pub enum CellState {
-    #[default]
     Dead,
     Alive,
 }
 
-pub type Grid = [[CellState; GRID_WIDTH]; GRID_HEIGHT];
+pub type Grid = Vec<Vec<CellState>>;
 
-fn valid_coordinate(x: usize, y: usize) -> bool {
-    x < GRID_WIDTH && y < GRID_HEIGHT
+fn valid_coordinate(grid: &Grid, row: usize, col: usize) -> bool {
+    let (width, height) = dimensions(grid);
+    row < height && col < width
 }
 
-pub fn get_cell_state(grid: &Grid, x: usize, y: usize) -> Option<CellState> {
-    if valid_coordinate(x, y) {
-        Some(grid[x][y])
-    } else {
-        None
-    }
+pub fn get_cell_state(grid: &Grid, x: usize, y: usize) -> CellState {
+    grid.get(x)
+        .and_then(|row| row.get(y))
+        .map_or(CellState::Dead, |s| *s)
```

Comment:
I think this can be changed by a `.copied().unwrap_or(Cellstate::Dead)`

Reply:
Done